enhancements/operator-conditions.md

```diff
@@ -0,0 +1,188 @@
+
+The `Upgradeable` condition might be useful when:
+
+- An operator is about to start a critical process and should not be upgraded until after the process is completed.
+- The operator is performing a migration of CRs that must be completed before the operator is ready to be upgraded.
+
+##### Example Condition with the Upgradeable Condition
+
+```yaml
+apiVersion: operators.coreos.com/v1
+kind: Condition
+metadata:
+  name: foo-operator-zdgen-asdf # RandomGen Name
+  namespace: operators
+status:
+   conditions:
+    type: Upgradeable # The name of the `foo` OLM Supported Condition.
+    status: "False"   # The operator is not ready to be upgraded.
+    reason: "migration"
+    message: "The operator is performing a migration."
+    lastTransitionTime: "2020-08-24T23:15:55Z"
+```
+
+Given that the `Upgradable Condition`'s status is set to `False`, OLM will understand that it should not upgrade the operator.
+
+### Non-OLM Supported Conditions
+
+There may be instances where an Operator may want to communicate a condition to users that isn't known by OLM or expected to change OLM behavior. Operators may communicate these "Non-OLM Supported Conditions" by appending the conditions to the `status.conditions` array of the `Condition` CR provided that the `Type` of the condition does not intersect with any "OLM Supported Condition" `Types`.
+
+### Overriding an OLM Supported Condition
+
+There are times as a Cluster Admin that you may want to ignore an "OLM Supported Condition" reported by an Operator. For example, imagine that a known version of an operator always communicates that it is not upgradeable. In this instance, you may want to upgrade the operator despite the operator communicating that it is not upgradeable. This could be accomplished by overriding the `OLM Supported Condition` by adding the condition's name to the `spec.overrides` array in the `Condition` CR:
+
+```yaml
+apiVersion: operators.coreos.com/v1
+kind: Condition
+metadata:
+  name: foo-operator-zdgen-asdf # RandomGen Name
+  namespace: operators
+spec:
+  overrides:
+    - Upgradeable # Informs OLM to ignore the Upgradeable "OLM Supported Condition".
+    - Foo # Informs OLM to ignore the foo "OLM Supported Condition".
+status:
+   conditions:
+    type: Upgradeable
+    status: "False"
+    reason: "migration"
+    message: "The operator is performing a migration."
+    lastTransitionTime: "2020-08-24T23:15:55Z"
+```
+
+Since the `Upgradeable` condition is being overridden, OLM will allow the operator to be upgraded despite the `Upgradeable Condition`'s status being set to `False`.
+
+### Managing the Condition CR
+
+As outlined earlier in this enhancement, the Operator SDK team will provide a library that operators can use to manage conditions on their `Condition` CR. The functions provided within the library will detect if the `Condition` CRD is present on the cluster using the Discovery API. If the API is not present, the functions will perform no-ops.
```

Comment:
I would expect if I try to set a condition and the CRD is not available, that the functions would throw an error that I can catch and decide to ignore if it is not important.

Reply:
This is something that @joelanford and I discussed a few weeks ago at a high level in an attempt to set expectations for the library that the SDK project will develop. This is a great point, and something that should be considered when defining the specifics of the library in a follow-up PR to this enhancement.

Reply:
@jamstah the `Ready` condition is now being proposed in #51.